### PR TITLE
feat: Created Coins/Converter section of landing page

### PR DESCRIPTION
**Pull request by THEjacob1000**
## Key Features

- **Selectable Page Types**: Introduced a selector enabling users to toggle between different page views, specifically "coins" and "converter" types, enhancing the application's versatility and user engagement.
- **Updated Store Management**: Adjusted the Zustand store to include `pageType` variables, allowing for more dynamic and responsive UI states based on user interaction.
- **Coin Carousel**: Implemented a coin-carousel feature, empowering users to easily browse and select different cryptocurrencies for comparison or data inspection, improving the overall user experience.
- **Bitcoin Price and Volume Graphs**: Developed interactive graphs to display Bitcoin's price and volume data over time, offering users insightful and detailed market analysis.
- **Dynamic Graph Data**: Enhanced the application to dynamically update graph data based on the selected cryptocurrency, providing a tailored analytical view to the user's specific interests.
- **Customizable Time Frame**: Users now have the ability to select the desired time frame for data display on graphs, with options extending up to 1 month, granting users control over the granularity of the data presented.

## Key Flaw

- **Limited API Data for Volume Chart**: The current API route integrated for fetching cryptocurrency data does not provide comprehensive volume information for cryptocurrencies other than Bitcoin. As a result, the volume chart's responsiveness is constrained to Bitcoin data only. This limitation affects the application's ability to display equivalent volume data for other selected cryptocurrencies, potentially impacting user experience for those interested in assets beyond Bitcoin.

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,3 +1,183 @@
...
+                      <button
+                        key={index}
+                        className={`relative w-full font-semibold z-10 ${
+                          index === timeframe
+                            ? "text-primary-foreground"
+                            : ""
+                        }`}
+                        onClick={() => {
+                          setTimeframe(index);
+                          console.log(timeframe);
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx]: no console

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,3 +1,183 @@
...
+                        timeframe === 2 &&
+                          "left-full -translate-x-[200%]",
+                        timeframe === 3 &&
+                          "left-full -translate-x-[100%]"
+                      )}
+                      aria-hidden="true"
+                    ></div>
+                    {timeframes.map((tf, index) => (
+                      <button
+                        key={index}
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx]: don;t use index as key

File: src/components/ConvertSearchbar.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                      updatedSelectedCoins[1] = newCoin.id;
+                    }
+                  }
+
+                  useCryptoStore
+                    .getState()
+                    .setSelectedCoins(updatedSelectedCoins);
+
+                  setOpen(false);
+                }
```
Comment (yoloOnTheBattlefield) [inline src/components/ConvertSearchbar.tsx]: put this code in a function

File: src/components/PricesChart.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            startDate = subDays(startOfToday(), 7);
+            break;
+          case "14D":
+            startDate = subDays(startOfToday(), 14);
+            break;
+          case "1M":
+            startDate = subMonths(startOfToday(), 1);
+            break;
+          default:
+            startDate = subMonths(startOfToday(), 1);
```
Comment (yoloOnTheBattlefield) [inline src/components/PricesChart.tsx]: you repeat startOfToday 5 times, put that in a variable

File: src/components/VolumeChart.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+          startDate = subDays(startOfToday(), 7);
+          break;
+        case "14D":
+          startDate = subDays(startOfToday(), 14);
+          break;
+        case "1M":
+          startDate = subMonths(startOfToday(), 1);
+          break;
+        default:
+          startDate = subMonths(startOfToday(), 1);
```
Comment (yoloOnTheBattlefield) [inline src/components/VolumeChart.tsx]: same here, you could extract this whole logic in a function

File: src/components/VolumeChart.tsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            plugins: {
+              legend: {
+                display: false,
+              },
+              tooltip: {
+                enabled: true,
+              },
+            },
+            responsive: true,
+            maintainAspectRatio: false,
```
Comment (yoloOnTheBattlefield) [inline src/components/VolumeChart.tsx:123]: you could save this options object and reuse it

File: src/components/ui/carousel.tsx [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+          scrollPrev()
+        } else if (event.key === "ArrowRight") {
+          event.preventDefault()
+          scrollNext()
+        }
+      },
+      [scrollPrev, scrollNext]
+    )
+
+    React.useEffect(() => {
```
Comment (yoloOnTheBattlefield) [inline src/components/ui/carousel.tsx]: useEffect, not React.useEffect

File: src/lib/store.ts [hunk 8]
```diff
@@ -1,30 +1,73 @@
...
+          ? new Error("Exceeded maximum selected coins")
+          : [...state.selectedCoin, coin];
+        if (newSelectedCoin instanceof Error) {
+          throw new Error("Exceeded maximum selected coins");
+        }
+        return {
+          selectedCoin: newSelectedCoin,
+        };
+      } else {
+        console.log(
```
Comment (yoloOnTheBattlefield) [inline src/lib/store.ts]: no console

File: src/lib/store.ts [hunk 9]
```diff
@@ -1,30 +1,73 @@
...
+  selectedCoin: ["bitcoin"],
+  changeSelectedCoin: (coin) =>
+    set((state) => {
+      if (state.compare) {
+        const isSelected = state.selectedCoin.includes(coin);
+        console.log(
+          isSelected
+            ? state.selectedCoin.filter((id) => id !== coin)
+            : [...state.selectedCoin, coin]
+        );
```
Comment (yoloOnTheBattlefield) [inline src/lib/store.ts]: no console

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/app/page.tsx (hunk 2 above)
Comment (THEjacob1000) [reply src/app/page.tsx]: changed to just use timeframe as the key

File: src/app/page.tsx (hunk 1 above)
Comment (THEjacob1000) [reply src/app/page.tsx]: commented out console call

File: src/components/ConvertSearchbar.tsx (hunk 3 above)
Comment (THEjacob1000) [reply src/components/ConvertSearchbar.tsx]: handleSelectCoin created and implemented

File: src/components/PricesChart.tsx (hunk 4 above)
Comment (THEjacob1000) [reply src/components/PricesChart.tsx]: stored in variable todayDate

File: src/components/VolumeChart.tsx (hunk 5 above)
Comment (THEjacob1000) [reply src/components/VolumeChart.tsx]: function created in @/lib/utils.ts and implemented in both files

File: src/lib/store.ts (hunk 9 above)
Comment (THEjacob1000) [reply src/lib/store.ts]: removed console logging

File: src/lib/store.ts (hunk 8 above)
Comment (THEjacob1000) [reply src/lib/store.ts]: removed console log

File: src/components/ui/carousel.tsx (hunk 7 above)
Comment (THEjacob1000) [reply src/components/ui/carousel.tsx]: updated file to import each component individually

**Pull request merged**